### PR TITLE
Add toggle for prognostic TKE and fix TKE advection

### PR DESCRIPTION
**Pull request by dennisYatunin**
## Purpose 

This PR adds the `--prognostic_tke` command-line option, which currently just toggles the advection of TKE. It also adds the vertical advection of TKE and cleans up the advection tendency functions. Related to #1916.

## Content

- Add a boolean type parameter to `EDMFX` and `DiagnosticEDFMX`, and set it to `parsed_args["prognostic_tke"]`.
- Define the `use_prognostic_tke` function to extract this type parameter (or to return `false` when there is no `turbconv_model`).
- Modify `horizontal_advection_tendency!` as follows:
    - Unify the horizontal advection of TKE for `EDMFX` and `DiagnosticEDFMX`.
    - Ensure that TKE is only advected when `use_prognostic_tke` returns `true`.
- Modify `explicit_vertical_advection_tendency!` as follows:
    - Extract all cached values at the beginning of this function.
    - Wrap all vertical operations in calls to `bycolumn`. Since there are two sets of vertical operations separated by horizontal operations, two calls to `bycolumn` are needed.
    - Add the vertical advection of TKE.
    - Ensure that TKE is only advected when `use_prognostic_tke` returns `true`.
    - Add comments to indicate which parts of this function will need to be moved to `implicit_vertical_advection_tendency!`.
- Add `--prognostic_tke true` to the `diagnostic_edmfx_aquaplanet` job. Since the initial TKE is 0 for this job, the simulation will not crash when TKE is advected.
- Remove `ᶜρa⁰` from the cache for `DiagnosticEDMFX` because it is not being used (we approximate it with `Y.c.ρ`).

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by dennisYatunin** [comment]
bors try

File: src/prognostic_equations/advection.jl [hunk 1]
```diff
@@ -85,8 +81,17 @@ function explicit_vertical_advection_tendency!(Yₜ, Y, p, t)
     (; ᶜu, ᶠu³, ᶜK, ᶜf) = p
     ᶜω³ = p.ᶜtemp_CT3
     ᶠω¹² = p.ᶠtemp_CT12
+    if p.atmos.turbconv_model isa AbstractEDMF
+        (; ᶠu³⁰, edmfx_upwinding) = p
+        (; dt) = p.simulation
+        ᶜρa⁰ = p.atmos.turbconv_model isa DiagnosticEDMFX ? Y.c.ρ : p.ᶜρa⁰
```
Comment (szy21) [inline src/prognostic_equations/advection.jl:87]: Maybe this breaks the allocation?

File: src/prognostic_equations/advection.jl [hunk 2]
```diff
@@ -172,6 +181,19 @@ function explicit_vertical_advection_tendency!(Yₜ, Y, p, t)
...
         end
+
+        # TODO: Move this to implicit_vertical_advection_tendency!.
+        if use_prognostic_tke(p.atmos.turbconv_model)
+            vertical_transport!(
+                Yₜ.c.sgs⁰.ρatke[colidx],
+                ᶜJ[colidx],
+                ᶜρa⁰[colidx],
+                ᶠu³⁰[colidx],
+                Y.c.sgs⁰.ρatke[colidx],
```
Comment (szy21) [inline src/prognostic_equations/advection.jl:192]: This should be `tke`, not `ρatke` I think.

**Comment by dennisYatunin** [comment]
bors try

**Comment by dennisYatunin** [comment]
bors try

**Comment by dennisYatunin** [comment]
bors try

**Comment by dennisYatunin** [comment]
bors try

**Pull request merged**